examples/src/main/java/com/datastax/oss/driver/examples/mapper/KillrVideoMapperExample.java

```diff
@@ -0,0 +1,148 @@
+ *
+ * <p>Preconditions:
+ *
+ * <ul>
+ *   <li>An Apache Cassandra(R) cluster is running and accessible through the contacts points
+ *       identified by basic.contact-points (see application.conf).
+ *   <li>that cluster has a {@code killrvideo} keyspace with the KillrVideo tables (see {@code
+ *       src/main/resources/killrvideo_schema.cql} in this project).
+ *   <li>the tables are empty.
+ * </ul>
+ *
+ * <p>Side effects:
+ *
+ * <ul>
+ *   <li>inserts data in the tables.
+ * </ul>
+ *
+ * @see <a href="https://docs.datastax.com/en/developer/java-driver/latest/manual/mapper/">Java
+ *     Driver Mapper manual</a>
+ */
+public class KillrVideoMapperExample {
+
+  private static final CqlIdentifier KEYSPACE_ID = CqlIdentifier.fromCql("killrvideo");
+
+  public static void main(String[] args) {
+    try (CqlSession session = CqlSession.builder().build()) {
+      KillrVideoMapper mapper =
+          KillrVideoMapper.builder(session).withDefaultKeyspace(KEYSPACE_ID).build();
+
+      // Create a new user
+      UserDao userDao = mapper.userDao();
+
+      Optional<User> maybeUser =
+          userDao.create("test", "user", "testuser@example.com", "password123".toCharArray());
+      if (!maybeUser.isPresent()) {
+        throw new AssertionError("User creation should succeed (expected empty tables)");
+      }
+      User user = maybeUser.get();
+      System.out.println("Created " + user);
+
+      // Creating another user with the same email should fail
+      assert !userDao
+          .create("test2", "user", "testuser@example.com", "secret123".toCharArray())
+          .isPresent();
+
+      // Simulate login attempts
+      tryLogin(userDao, "testuser@example.com", "password123");
+      tryLogin(userDao, "testuser@example.com", "secret123");
+
+      // Insert a video
+      VideoDao videoDao = mapper.videoDao();
+
+      Video video = new Video();
+      video.setUserid(user.getUserid());
+      video.setName(
+          "Getting Started with DataStax Apache Cassandra as a Service on DataStax Constellation");
+      video.setLocation("https://www.youtube.com/watch?v=68xzKpcZURA");
+      Set<String> tags = new HashSet<>();
+      tags.add("apachecassandra");
+      tags.add("nosql");
+      tags.add("hybridcloud");
+      video.setTags(tags);
+
+      videoDao.create(video);
+      System.out.printf("Created video [%s] %s%n", video.getVideoid(), video.getName());
+
+      // Check that associated denormalized tables have also been updated:
+      PagingIterable<UserVideo> userVideos = videoDao.getByUser(user.getUserid());
+      System.out.printf("Videos for %s %s:%n", user.getFirstname(), user.getLastname());
+      for (UserVideo userVideo : userVideos) {
+        System.out.printf("  [%s] %s%n", userVideo.getVideoid(), userVideo.getName());
+      }
+
+      String yyyymmdd =
+          DateTimeFormatter.ofPattern("yyyyMMdd").withZone(ZoneOffset.UTC).format(Instant.now());
```

Comment:
That `Instant.now()` is a bit non-intuitive. It is also embedded within the logic of `CreateVideoQueryProvider`.
To improve design maybe we could refactor it to use a time provider? Instead of hardcoding that in the logic?

Reply:
Yeah I admit it looks a bit weird in the middle of the client code. I'll extract it in a method and add a few explanations.